src/Mod/Fem/Gui/TaskCreateElementSet.cpp

```diff
@@ -0,0 +1,820 @@
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+#include <QString>
+#include <QSlider>
+#include <Python.h>
+#include <memory>
+
+#include <string.h>
+#include <stdio.h>
+#include <regex>
+
+#include <Standard_math.hxx>
+
+#include <Inventor/nodes/SoEventCallback.h>
+#include <Inventor/nodes/SoCamera.h>
+#include <Inventor/events/SoMouseButtonEvent.h>
+
+#include <SMESH_Mesh.hxx>
+#include <SMESHDS_Mesh.hxx>
+#include <SMDSAbs_ElementType.hxx>
+
+#endif
+
+#include <ui_TaskCreateElementSet.h>
+#include "TaskCreateElementSet.h"
+#include <QMessageBox>
+#include <Gui/Command.h>
+#include <Gui/Application.h>
+#include <Gui/Document.h>
+#include <Gui/BitmapFactory.h>
+#include <Gui/ViewProvider.h>
+#include <Gui/WaitCursor.h>
+#include <Base/Console.h>
+#include <Gui/View3DInventor.h>
+#include <Gui/View3DInventorViewer.h>
+#include <Gui/Utilities.h>
+#include <Gui/MainWindow.h>
+#include <App/Document.h>
+#include <SMESH_Mesh.hxx>
+#include <SMESHDS_Mesh.hxx>
+#include <SMDSAbs_ElementType.hxx>
+
+#include <Mod/Fem/App/FemMesh.h>
+#include <Mod/Fem/App/FemMeshObject.h>
+#include <Mod/Fem/App/FemSetElementNodesObject.h>
+#include "ViewProviderFemMesh.h"
+#include "FemSelectionGate.h"
+using namespace FemGui;
+using namespace Gui;
+using namespace Fem;
+using namespace Base;
+using namespace boost;
+
+std::string inp_file = "abaqusOutputFileEraseElements.inp";
+
+std::string createdMesh = "cDUMMY";
+std::string startResultMesh = "StartResultMesh"; // StartResultMesh";
+std::string newResultMesh = "NewResultMesh";
+//std::string resultMeshName = "ResultMesh";
+std::string newFemMesh = "NewFemMesh";
+
+std::string TaskCreateElementSet::currentProject = "";
+
+std::string uniqueMesh = "";
+std::string newProject = "";
+std::string resultMesh = "ResultMesh";
+std::string lastName = "";
+std::string highLightMesh; //  name of highlighted mesh
+std::string meshType;      // type of - either 'result' or 'femmesh'
+int passResult = 0;
+int passFemMesh = 0;
+int maxProject = 10;
+double **nodeCoords; // these are node coords
+int *nodeNumbers;    // these are node numbers - probably not required[100];
+
+void addFaceToMesh(const std::vector<const SMDS_MeshNode *> nodes, SMESHDS_Mesh *MeshDS, int EID)
+{
+    int nbNodes = nodes.size();
+    switch (nbNodes)
+    {
+    case 3:
```

Comment:
This is hardly readable. What is case 3 etc? Please use an enum instead. Then you get a self-explaining name like e.g. "case Nodes::Volume" of whatever the different cases stand for.